### ISSUE TITLE
Request:  Future update for multiple Rokus

### ISSUE THREAD

**Issue by jeffreyjue1** [bschlenk/homebridge-roku#5]
As discussed earlier, one of the future major updates should be able to continue scanning after finding the first Roku for households that have more than 1.  When I got the plugin to work, I was confused as to why the TCL Roku TV wasn't controlled and Roku 2 in another was being controlled until I realized that the setup scan was for the first IP address scanned.  The current work around is to unplug other Rokus one by one until all configurations are collected.

However, the update to better handle TV controls rather than "turn on" or "turn off" some of the commands should come first.

**Comment by bschlenk** [comment]
It looks like someone [created a fork](https://github.com/tonym97/homebridge-roku-client) that supports multiple devices. I will look into copying their changes once the iOS 12.2 stuff has been released. iOS 12.2 will have native support for TVs and will show up as one device instead of the many devices homebridge-roku currently shows up as, which would simplify things when having multiple Roku devices.

**Comment by NateUT99** [comment]
I hate to bring up such an old issue, but did support for multiple Rokus on the same network ever get integrated into the 3.0.0-alpha?

**Comment by ncovercash** [comment]
I'd at least love it if there was a way to pass the IP to `homebridge-roku-config` – that way on a network with multiple rokus, the desired one could be chosen.

**Comment by DennisFury** [comment]
Any chance of getting multiple TV support?  I've got 3 Roku TVs and 3 Samsung TVs, the Samsung home bridge client supports multiple devices and I can use the built in Apple remote to control any of them.  Would be great to do the same on Roku...

**Comment by RaziAmin** [comment]
Hi there, I'm just bumping this feature hoping to get an update for it

**Comment by hadamlenz** [comment]
incase anyone comes to this page looking for a solution.  in order to pair to more than one roku you can only have one roku on the network at a time

**Comment by DennisFury** [comment]
> 
> 
> incase anyone comes to this page looking for a solution. in order to pair to more than one roku you can only have one roku on the network at a time. then pair. then you can add all the rokus back and they are all present in homekit

Yeah, but that only adds a non-TV device in Homekit that allows us to turn off/on TVs.  If you look at your Apple Remote, you can only control up/down/left/right/select/back with one Roku TV.

We're wanting to get all Roku TVs on our network into the Apple Remote so we can actually control the TV.

**Comment by DennisFury** [comment]
Roku solved this for us...
https://support.roku.com/article/360057488733

The later Roku TVs are now homekit compatible and we can get them injected into the remote app, and they also have airplay 2 support.  My only older Roku TV not compatible is being taken care of by this plugin, with the single Roku TV it supports.  I got lucky.

**Comment by bschlenk** [comment]
I released a beta of version 4.0.0, you can install with `npm install -g homebridge-roku@next`. It supports multiple devices, they’ll all show up in the remote app, and they’re auto discovered now. Although you have to manually connect them following https://github.com/homebridge/homebridge/wiki/Connecting-Homebridge-To-HomeKit#how-to-add-homebridge-cameras--tvs

You can read about what’s changed [here](https://github.com/bschlenk/homebridge-roku/blob/4.0/CHANGELOG.md#400-beta0-2021-03-02).

**Comment by addisonjames** [comment]
Hey @bschlenk - Nevermind my comment before. The easiest way for me to install the beta was from the Homebridge UI under Install Alternate Version.
